src/jsonschema.rs

```diff
@@ -0,0 +1,219 @@
+use serde_json::Value;
+use std::collections::HashMap;
+
+/// The type enumeration does not contain any data and is used to determine
+/// available fields in the flattened tag. In JSONSchema parlance, these are
+/// known as `simpleTypes`.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+enum Type {
+    Null,
+    Boolean,
+    Number,
+    Integer,
+    String,
+    Object,
+    Array,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+enum AdditionalProperties {
+    True,
+    False,
+    Object(Box<Tag>),
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+struct Object {
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
This serde attribute will skip this field when serializing if it doesn't exist -- by default this field is instantiated to None. 

https://serde.rs/field-attrs.html#skip_serializing_if